Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -627,6 +629,58 @@ public IEnumerator TestPressableToggleHoloLens2()
             GameObject.Destroy(interactable.gameObject);
         }
 
+        [UnityTest]
+        /// <summary>
+        /// Test InteractableToggleCollection CurrentIndex updates
+        /// </summary>
+        public IEnumerator TestInteractableToggleCollectionIndexUpdate()
+        {
+            InteractableToggleCollection interactableToggleCollection;
+            int numRadials = 6;
+
+            AssembleInteractableToggleCollection(
+                out interactableToggleCollection,
+                numRadials,
+                Vector3.forward);
+
+            var toggleList = interactableToggleCollection.ToggleList;
+
+            int[] onClickEventCalled = new int[numRadials];
+
+            int index = 0;
+            foreach (var radial in toggleList)
+            {
+                int indexRadial = index;
+                radial.OnClick.AddListener(() => { onClickEventCalled[indexRadial] = 1; });
+                index++;
+            }
+
+            for (int j = 0; j < numRadials; j++)
+            {
+                interactableToggleCollection.CurrentIndex = j;
+                yield return null;
+
+                // If the CurrentIndex is changed the toggle should be visually updated and events should be triggered
+                for (int i = 0; i < numRadials; i++)
+                {
+                    if (i == interactableToggleCollection.CurrentIndex)
+                    {
+                        Assert.True(interactableToggleCollection.ToggleList[i].IsToggled);
```

Comment:
simplify to 

assert.areequals(interactableToggleCollection.ToggleList[i].IsToggled, i == interactableToggleCollection.CurrentIndex)
or
bool shouldBeSelected = i == interactableToggleCollection.CurrentIndex;
Assert.AreEquals(shouldBeSelected, interactableToggleCollection.ToggleList[i].IsToggled)

for the second assert
bool expectedClickCount = i <= j ? 1 : 0;
Assert.AreEquals(onClickEventCalled[i], expectedClickCount);

Reply:
Simplified